crates/cargo-contract/src/cmd/extrinsics/remove.rs

```diff
@@ -0,0 +1,242 @@
+            (None, None) => {
+                // default to the target contract wasm in the current project,
+                // inferred via the crate metadata.
+                load_code(&crate_metadata.dest_wasm)
+            }
+            (None, Some(code_hash)) => Ok(Code::Existing(*code_hash)),
+        }?;
+
+        let code_hash = contract_metadata.source.hash;
+
+        async_std::task::block_on(async {
+            let url = self.extrinsic_opts.url_to_string();
+            let client = OnlineClient::from_url(url.clone()).await?;
+            if self.extrinsic_opts.dry_run {
+                match self.remove_code_rpc(code, &client, &signer).await? {
+                    Ok(result) => {
+                        let remove_result = RemoveDryRunResult {
+                            result: String::from("Success!"),
+                            code_hash: format!("{:?}", result.code_hash),
+                        };
+                        if self.output_json {
+                            println!("{}", remove_result.to_json()?);
+                        } else {
+                            remove_result.print();
+                        }
+                    }
+                    Err(err) => {
+                        let metadata = client.metadata();
+                        let err = ErrorVariant::from_dispatch_error(&err, &metadata)?;
+                        if self.output_json {
+                            return Err(err);
+                        } else {
+                            name_value_println!("Result", err);
+                        }
+                    }
+                }
+                Ok(())
+            } else if let Some(code_stored) = self
+                .remove_code(&client, sp_core::H256(code_hash.0), &signer, &transcoder)
+                .await?
+            {
+                let remove_result = RemoveResult {
+                    code_hash: format!("{:?}", sp_core::H256(code_hash.0)),
+                };
+                if self.output_json {
+                    println!("{}", remove_result.to_json()?);
+                } else {
+                    remove_result.print();
+                }
+                Ok(())
+            } else {
+                Err(anyhow::anyhow!(
+                    "This contract with code hash: {:?} does not exist",
+           code_hash
+                )
+                .into())
+            }
+        })
+    }
+
+    async fn remove_code_rpc(
+        &self,
+        code: Code,
+        client: &Client,
+        signer: &PairSigner,
+    ) -> Result<CodeUploadResult<CodeHash, Balance>> {
+        let url = self.extrinsic_opts.url_to_string();
+        let token_metadata = TokenMetadata::query(client).await?;
+        let call_request = CodeRemoveRequest {
+            origin: signer.account_id().clone(),
+            code,
+        };
+        state_call(&url, "ContractsApi_remove_code", call_request).await
```

Comment:
I guess in order for this dry run to work we'd have to add `remove_code` to the runtime api: https://github.com/paritytech/substrate/blob/f465fee723c87b7348839b5140135b17d0a48743/frame/contracts/src/lib.rs#L1215